packages/workato-adapter/src/adapter.ts

```diff
@@ -140,14 +142,54 @@ export default class WorkatoAdapter implements AdapterOperations {
    */
   @logDuration('deploying account configuration')
   // eslint-disable-next-line class-methods-use-this
-  async deploy(): Promise<DeployResult> {
-    throw new Error('Not implemented.')
+  async deploy({ changeGroup, accountToServiceNameMap }: DeployOptions): Promise<DeployResult> {
+    // resolving workato and cross-service
+    if (changeGroup.groupID !== DEPLOY_USING_RLM_GROUP) {
+      throw new Error('not implemented')
+    }
+
+    const resolvedChanges = await Promise.all(
+      changeGroup.changes.map(change => resolveChangeElement(change, getLookUpNameFuncsToFunc([
+        ...getCrossServiceLookUpNameFuncs(accountToServiceNameMap),
+        workatoLookUpName,
+      ]), resolveValuesRecipeCodeWrapper))
+    )
+
+    const runner = this.createFiltersRunner()
+    await runner.preDeploy(resolvedChanges)
+
+    const [instanceChanges, nonInstanceChanges] = _.partition(resolvedChanges, isInstanceChange)
+    if (nonInstanceChanges.length > 0) {
+      log.warn(`We currently can't deploy types. Therefore, the following changes will not be deployed: ${
+        nonInstanceChanges.map(elem => getChangeData(elem).elemID.getFullName()).join(', ')}`)
+    }
+
+    const deployResult = await RLMDeploy(instanceChanges, this.client)
+
+    const appliedChangesBeforeRestore = [...deployResult.appliedChanges]
+    await runner.onDeploy(appliedChangesBeforeRestore)
+
+    const appliedChangesNameBeforeRestore = appliedChangesBeforeRestore
+      .map(change => getChangeData(change).elemID.getFullName())
+    const appliedChanges = changeGroup.changes.filter(change => {
+      const changeData = getChangeData(change)
+      return appliedChangesNameBeforeRestore.includes(isInstanceChange(change)
+      && isFromType([RECIPE_CODE_TYPE])(change)
+        ? getParent(changeData).elemID.getFullName()
+        : changeData.elemID.getFullName())
+    })
```

Comment:
there is some hidden logic in unexpected places here which makes this harder to follow, maybe something like this?
```suggestion
    const appliedChanges = changeGroup.changes.filter(change => {
      const changeData = getChangeData(change)
      const changeID = isInstanceChange(change) && isFromType([RECIPE_CODE_TYPE])(change)
        ? getParent(changeData).elemID.getFullName()
        : changeData.elemID.getFullName()
      return appliedChangesNameBeforeRestore.has(changeID)
    })
```